index.js

```diff
@@ -7,24 +7,39 @@ const fs = require('fs');
 
 async function run() {
   try {
-    const msys2base = await tc.downloadTool('http://repo.msys2.org/distrib/x86_64/msys2-base-x86_64-20190524.tar.xz');
-    const tar = await io.which('tar', true);
+    if (process.platform != 'win32') {
+      core.setFailed("MSYS2 does not work on non-windows platforms; please check the 'runs-on' field of the job");
+    }
 
-    const dest = path.join(process.env['RUNNER_TEMP'], 'msys')
+    const tar = await io.which('tar', true);
 
-    // For some reason, GNU Tar on Windows expects paths to be slash-separated
-    const normalizedDest = dest.replace(/\\/g, '/')
+    const tmp_dir = process.env['RUNNER_TEMP']
+    if (!tmp_dir) {
+      core.setFailed('environment variable RUNNER_TEMP is undefined');
+      return
+    }
+    const dest = path.join(tmp_dir, 'msys')
 
     await io.mkdirP(dest)
 
-    await exec.exec(`"${tar}"`, ['-x', '-J', '--force-local', '-C', normalizedDest, '-f', msys2base])
+    await exec.exec(`"${tar}"`, [
+      '-x', '-J', '--force-local',
+      // For some reason, GNU Tar on Windows expects paths to be slash-separated
+      '-C', dest.replace(/\\/g, '/'),
+      '-f', await tc.downloadTool('http://repo.msys2.org/distrib/x86_64/msys2-base-x86_64-20190524.tar.xz')
```

Comment:
I kinda don't like having the "meaty" part of the recipe (downloading the tarball) being done as a parameter to `exec`.